docs/clusterobservability.md

```diff
@@ -158,29 +167,202 @@ spec:
       enabled: true
 ```
 
-* In the above example the tracing and logging capabilities will use `minio` secret for connecting to the object storage.
-* The controller transforms the `minio` secret into secrets required by the `LokiStack` and `TempoStack` instances. For example, the `TempoStack` requires fields `bucket` and `LokiStack` uses the field `bucketnames`.
+* In the above example the tracing and logging capabilities will use S3 as object storage.
+* The controller transforms the configuration in `s3` secret into secrets required by the `LokiStack` and `TempoStack` instances.
 
 The various storage configuration per capability can be achieved by multiple `ClusterObservability` CRs, each with its own storage configuration.
 
 ### Object storage types
 
-Each object storage type has its own set of required fields which are configured in the secret.
-Object storage types will be added separately, there are plans to support the all the storage types supported by the capabilities.
+Each object storage type has its own set of required fields which are configured directly in the CR.
+There are plans to support all storage types required by the capabilities.
+
+#### Design principles
 
-#### S3 secret
+The `ClusterObservability` object storage configuration is directly held in the CR and secrets are used only to reference sensitive data.
+
+##### Alternative thanos-io/objstore
+
+The https://github.com/thanos-io/objstore client is used by Loki and Thanos to connect to the object storage.
+It offers a unified interface for different object storage types with a common configuration file.
+
+At the time of writing this document, the COO does not support Thanos, Loki operator does not directly support the `thanos-io/objstore`
+configuration file and the Tempo does not support it at all.
+
+#### Amazon S3 S3 / MinIO
+
+Supported by Tempo and Loki.
 
 ```yaml
-kubectl create secret generic s3 \
+spec:
+  storage:
+    objectStorage:
+      s3:
```

Comment:
@pavolloffay For STS we only need bucket, roleARN and region. The rest of the fields are not required. 
% oc create -f coo.yaml 
The ClusterObservability "tracing" is invalid: 
* spec.storage.objectStorage.s3.accessKeyID: Required value
* spec.storage.objectStorage.s3.accessKeySecret: Required value
* spec.storage.objectStorage.s3.endpoint: Required value